compiler/aarch64/codegen/ARM64SystemLinkage.cpp

```diff
@@ -286,6 +286,7 @@ TR::ARM64SystemLinkage::mapStack(TR::ResolvedMethodSymbol *method)
    uint32_t stackIndex = 0;
    ListIterator<TR::AutomaticSymbol> automaticIterator(&method->getAutomaticList());
    TR::AutomaticSymbol *localCursor = automaticIterator.getFirst();
+   bool frameNeeded = (machine->getLinkRegisterKilled());
 
    stackIndex = 8; // [sp+0] is for link register
```

Comment:
I think we can stop reserving `[sp+0]` for the link register when `machine->getLinkRegisterKilled()` is `false`.

Reply:
Just to make sure I understand your suggestion, you're saying this first initialization of stackIndex (to 8) can also be made conditional upon the link register being killed (and otherwise initializing to 0) ?

Reply:
I decided to commit this change as it seemed right to me. If I misunderstood, I can always change it back or to something else :) . Change has now been pushed.

Reply:
Yes, your change is the same as what I was thinking of.